### PR TITLE
AWS Security Hub parser add unique_id_from_tool and deduplication

### PR DESCRIPTION
**Pull request by nlandais**
- Add name of the AWS resource in the title for ease of use and to help in the remediation process
- Dedupe finding on unique_id_from_tool: use the SecHub finding id (value after the last / from the finding's ARN)

### CODE REVIEW

File: dojo/urls.py [hunk 1]
```diff
@@ -213,7 +213,6 @@
     url(r'^%sapi/v2/api-token-auth/' % get_system_setting('url_prefix'), tokenviews.obtain_auth_token),
     url(r'^%sapi/v2/doc/' % get_system_setting('url_prefix'), schema_view.with_ui('swagger', cache_timeout=0), name='api_v2_schema'),
     url(r'^robots.txt', lambda x: HttpResponse("User-Agent: *\nDisallow: /", content_type="text/plain"), name="robots_file"),
-    url(r'^manage_files/(?P<oid>\d+)/(?P<obj_type>\w+)$', views.manage_files, name='manage_files'),
```
Comment (valentijnscholten) [inline dojo/urls.py:216]: mistake?

File: dojo/tools/awssecurityhub/parser.py [hunk 2]
```diff
@@ -36,41 +34,27 @@ def get_items(self, tree, test):
...
-
-    if finding.get('Compliance', {}).get('Status', "PASSED"):
-        if finding.get('LastObservedAt', None):
-            try:
-                mitigated = datetime.strptime(finding.get('LastObservedAt'), "%Y-%m-%dT%H:%M:%S.%fZ")
-            except:
-                mitigated = datetime.strptime(finding.get('LastObservedAt'), "%Y-%m-%dT%H:%M:%fZ")
-        else:
-            mitigated = datetime.utcnow()
-    else:
```
Comment (valentijnscholten) [inline dojo/tools/awssecurityhub/parser.py:59]: can you explain how this competes with close old findings? I think PASSED findings are still imported so will not be closed by close_old_findings?

**Review by valentijnscholten — CHANGES_REQUESTED** [review]
comments

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: dojo/urls.py (hunk 1 above)
Comment (nlandais) [reply dojo/urls.py]: Sorry,  this was a mistake and I've updated the PR

File: dojo/tools/awssecurityhub/parser.py (hunk 2 above)
Comment (nlandais) [reply dojo/tools/awssecurityhub/parser.py]: I've re-instated the code above. In my team's implementation/workflow we will not hit this path since we're using boto to filter the results and only report on new findings but the proposed change may negatively impact other implementations.

File: dojo/tools/awssecurityhub/parser.py [hunk 3]
```diff
@@ -36,41 +34,27 @@ def get_items(self, tree, test):
...
                       test=test,
-                      severity=severity,
                       description=description,
                       mitigation=mitigation,
                       references=references,
-                      cve=cve,
-                      cwe=cwe,
+                      severity=severity,
+                      active=False,
+                      verified=False,
```
[the remark this answers is not in the document]
Comment (nlandais) [reply dojo/tools/awssecurityhub/parser.py]: Resolved

File: .gitmodules [hunk 4]
```diff
@@ -1,3 +1,3 @@
 [submodule "docs/themes/docsy"]
 	path = docs/themes/docsy
-	url = git@github.com:google/docsy.git
+	url = https://github.com/google/docsy.git
```
Comment (damiencarol) [inline .gitmodules:3]: I don't think you should push this modifiction

**Review by damiencarol — CHANGES_REQUESTED** [review]
Could you add some checks in the unit tests that show the new data in the findings?

File: .gitmodules (hunk 4 above)
Comment (nlandais) [reply .gitmodules:3]: git@github.com bombs for me when building in CodeBuild (AWS) with an issue with key not present and access denied, so that's not cool ...

Comment (damiencarol) [reply .gitmodules:3]: you can keep your local modifications, it's just that this file is not linked to your pull request so... doesn't make sense to modify it :D

**Comment by nlandais** [comment]
> Could you add some checks in the unit tests that show the new data in the findings?
@damiencarol Bonjour Damien, pourrais-tu elaborer sur la requete consistant a modifier les unittests pour le filtre d'ingestion des donnees provenant de AWS SecurityHub ? 
The parser does not change much in functionality, the big change in this PR is the fact that the deduplication and filtering was not working properly when ingesting one report at a time from AWS SecurityHub. 
The behavior observed prior to this change request was as follows:
- When drilling down into to a finding , the "imilar findings" section showed all the other AWS Security Hub findings regardless  of the AWS resource associated with the finding.
- Each new imported finding marked the previous one as duplicate, which is consistent with the behavior reported in the first bullet point.
In conclusion, there isn't any new data in the findings, the proposed change implements the behavior we come to expect with DefectDojo when it comes to aggregating similar findings, managing duplicate findings and closing stale/old findings.
Feel free to DM me on the OWASP defectodjo Slack channel (handle: Nicolas Landais) to discuss further.
-Merci

File: dojo/tools/awssecurityhub/parser.py [hunk 5]
```diff
@@ -36,17 +35,15 @@ def get_items(self, tree, test):
 
 
 def get_item(finding, test):
+    finding_id = finding.get('Id', "").split('/')[-1]
```
Comment (damiencarol) [inline dojo/tools/awssecurityhub/parser.py:38]: I took a look at the data and I think your can keep the complete ARN.

**Review by damiencarol — CHANGES_REQUESTED** [review]
As you are adding modifications to `unique_id_from_tool` attribute, could you add some checks in the unit tests that checks these new data?

**Comment by damiencarol** [comment]
@nlandais there is something out of scope in the PR. it seems that one sub-module for documentation is updated.

File: dojo/tools/awssecurityhub/parser.py [hunk 6]
```diff
@@ -59,18 +56,16 @@ def get_item(finding, test):
...
+                      severity=severity,
+                      active=True,
+                      verified=False,
                       false_p=false_p,
-                      duplicate=duplicate,
-                      out_of_scope=out_of_scope,
-                      mitigated=mitigated,
-                      impact="No impact provided")
+                      unique_id_from_tool=finding_id,
+                      )
```
Comment (damiencarol) [inline dojo/tools/awssecurityhub/parser.py:69]: You removed `mitigated`. Did you really wanted to remove this feature?

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: dojo/tools/awssecurityhub/parser.py (hunk 6 above)
Comment (damiencarol) [reply dojo/tools/awssecurityhub/parser.py:69]: I think we should add back `mitigated` and `is_mitigated`

**Review by damiencarol — APPROVED** [review]

**Comment by damiencarol** [comment]
@valentijnscholten could you take another look, @nlandais made a lot of modifications since the first reviews.

**Review by valentijnscholten — APPROVED** [review]
